### PR TITLE
Add support for keyboard 'Bolt'

### PR DESCRIPTION
**Pull request by jackytrabbit**
Add support for keyboard 'Bolt'

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lazydesigners/bolt/config.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4C44 // "LD"
+#define PRODUCT_ID      0x0041
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    LAZYDESIGNERS
+#define PRODUCT         Bolt
+#define DESCRIPTION     A 40% ergo keyboard
```
Comment (zvecr) [inline keyboards/lazydesigners/bolt/config.h:28]:
```suggestion
```

File: keyboards/lazydesigners/bolt/rules.mk [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (zvecr) [inline keyboards/lazydesigners/bolt/rules.mk:26]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/lazydesigners/bolt/readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+A 40% ergo keyboard designed and produced by [LAZYDESIGNERS](http://lazydesigners.cn).
+
+Keyboard Maintainer: [LAZYDESIGNERS](https://github.com/jackytrabbit)
+
+Hardware Supported: Bolt
+
+Hardware Availability: Check [LAZYDESIGNERS's homepage.](http://lazydesigners.cn)
+
+To enter the bootloader, either push the RESET button on the PCB or the RESET button on the FN layer.
```
Comment (zvecr) [inline keyboards/lazydesigners/bolt/readme.md:13]:
```suggestion
# Bolt

![bolt](https://i.loli.net/2020/07/15/whZWXe1Il2cEftR.jpg)

A 40% ergo keyboard designed and produced by [LAZYDESIGNERS](http://lazydesigners.cn).

* Keyboard Maintainer: [LAZYDESIGNERS](https://github.com/jackytrabbit)
* Hardware Supported: Bolt
* Hardware Availability: Check [LAZYDESIGNERS's homepage.](http://lazydesigners.cn)

Make example for this keyboard (after setting up your build environment):

    make lazydesigners/bolt:default

Flashing example for this keyboard:

    make lazydesigners/bolt:default:flash

**Reset Key:** To enter the bootloader, either push the RESET button on the PCB or the RESET button on the FN layer.

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/lazydesigners/bolt/info.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (zvecr) [inline keyboards/lazydesigners/bolt/info.json:1]: This renders incorrectly (using `ctrl + shift + i` within configurator)
image

Comment (jackytrabbit) [reply keyboards/lazydesigners/bolt/info.json:1]: Thanks! Update info.json with a clear layout.

File: keyboards/lazydesigners/bolt/info.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    "keyboard_name": "Bolt",
+    "url": "http://lazydesigners.cn",
+    "maintainer": "LAZYDESIGNERS",
+    "width": 14.25,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 46,
```
Comment (fauxpark) [inline keyboards/lazydesigners/bolt/info.json:9]:
```suggestion
```

File: keyboards/lazydesigners/bolt/bolt.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
+	K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, \
+	K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, \
+	K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, \
+	K300, K301,       K303, K304,       K306, K307, K308, K309, K310, K311  \
```
Comment (noroadsleft) [inline keyboards/lazydesigners/bolt/bolt.h:26]: Rearranges the layout macro so the `MO(2)` key is actually on the ZXCV row.

```suggestion
	K000, K001, K002, K003, K004, K005,             K006, K007, K008, K009, K010, K011, \
	K100, K101, K102, K103, K104, K105,             K106, K107, K108, K109, K110, K111, \
	K200, K201, K202, K203, K204, K205,       K206, K207, K208, K209, K210, K211, K311, \
	K300, K301,             K303, K304,       K306, K307,             K308, K309, K310  \
```

File: keyboards/lazydesigners/bolt/info.json [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                {"label":"K300 (D5,F1)", "x":0, "y":3, "w":1.25},
+                {"label":"K301 (D5,F4)", "x":1.25, "y":3, "w":1.25},
+                {"label":"K303 (D5,F6)", "x":3.25, "y":3, "w":1.25},
+                {"label":"K304 (D5,F7)", "x":4.5, "y":3, "w":2.25},
+                {"label":"K306 (D5,B3)", "x":7.25, "y":3, "w":2.25},
+                {"label":"K307 (D5,B7)", "x":9.5, "y":3},
+                {"label":"K308 (D5,D0)", "x":11.25, "y":3},
+                {"label":"K309 (D5,D3)", "x":12.25, "y":3},
+                {"label":"K310 (D5,D2)", "x":13.25, "y":3},
+                {"label":"K311 (D5,D1)", "x":13.25, "y":2}
```
Comment (noroadsleft) [inline keyboards/lazydesigners/bolt/info.json:55]: Moves the `MO(2)` key object's place in the sequence.

```suggestion
                {"label":"K311 (D5,D1)", "x":13.25, "y":2},
                {"label":"K300 (D5,F1)", "x":0, "y":3, "w":1.25},
                {"label":"K301 (D5,F4)", "x":1.25, "y":3, "w":1.25},
                {"label":"K303 (D5,F6)", "x":3.25, "y":3, "w":1.25},
                {"label":"K304 (D5,F7)", "x":4.5, "y":3, "w":2.25},
                {"label":"K306 (D5,B3)", "x":7.25, "y":3, "w":2.25},
                {"label":"K307 (D5,B7)", "x":9.5, "y":3},
                {"label":"K308 (D5,D0)", "x":11.25, "y":3},
                {"label":"K309 (D5,D3)", "x":12.25, "y":3},
                {"label":"K310 (D5,D2)", "x":13.25, "y":3}
```

File: keyboards/lazydesigners/bolt/info.json [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+                {"label":"K008 (F0,D0)", "x":9.5, "y":0},
+                {"label":"K009 (F0,D3)", "x":10.5, "y":0},
+                {"label":"K010 (F0,D2)", "x":11.5, "y":0},
+                {"label":"K011 (F0,D1)", "x":12.5, "y":0, "w":1.75},
+                {"label":"K100 (C7,F1)", "x":0, "y":1, "w":1.25},
+                {"label":"K101 (C7,F4)", "x":1, "y":1},
+                {"label":"K102 (C7,F5)", "x":2, "y":1},
+                {"label":"K103 (C7,F6)", "x":3, "y":1},
+                {"label":"K104 (C7,F7)", "x":4, "y":1},
+                {"label":"K105 (C7,C6)", "x":5, "y":1},
```
Comment (noroadsleft) [inline keyboards/lazydesigners/bolt/info.json:27]: Visual alignment fix here. (You have some keys overlapped.)

```suggestion
                {"label":"K101 (C7,F4)", "x":1.25, "y":1},
                {"label":"K102 (C7,F5)", "x":2.25, "y":1},
                {"label":"K103 (C7,F6)", "x":3.25, "y":1},
                {"label":"K104 (C7,F7)", "x":4.25, "y":1},
                {"label":"K105 (C7,C6)", "x":5.25, "y":1},
```

File: keyboards/lazydesigners/bolt/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+#define LT2_TAB LT(2, KC_TAB)
+#define LT1_SPC LT(1, KC_SPC)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+               KC_GESC,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
+               LT2_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
+               KC_LSFT,  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,
+               KC_LCTL,  KC_LALT, KC_LGUI, LT1_SPC, KC_SPC,  LT1_SPC, KC_LEFT, KC_DOWN, KC_RGHT, MO(2)
```
Comment (noroadsleft) [inline keyboards/lazydesigners/bolt/keymaps/default/keymap.c:26]: Moving the `MO(2)` key...

```suggestion
               KC_GESC,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                      KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
               LT2_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                      KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
               KC_LSFT,  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,             KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   MO(2),
               KC_LCTL,  KC_LALT,                   KC_LGUI, LT1_SPC,          KC_SPC,  LT1_SPC,                   KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/lazydesigners/bolt/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+             ),
+    [1] = LAYOUT(
+               RESET,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_NO,
+               KC_INS,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
+               KC_NO,    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+               KC_VOLD,  KC_MUTE, KC_VOLU, KC_TRNS, KC_TRNS, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO
```
Comment (noroadsleft) [inline keyboards/lazydesigners/bolt/keymaps/default/keymap.c:32]:
```suggestion
               RESET,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_NO,
               KC_INS,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                     KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
               KC_NO,    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,           KC_F12,  KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
               KC_VOLD,  KC_MUTE,                   KC_VOLU, KC_TRNS,          KC_TRNS, KC_NO,                     KC_NO,   KC_NO,   KC_NO
```

File: keyboards/lazydesigners/bolt/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+             ),
+    [2] = LAYOUT(
+               RESET,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+               KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+               KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+               KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO
```
Comment (noroadsleft) [inline keyboards/lazydesigners/bolt/keymaps/default/keymap.c:38]:
```suggestion
               RESET,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,                     KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
               KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,                     KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
               KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,            KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
               KC_NO,    KC_NO,                     KC_NO,   KC_NO,            KC_NO,   KC_NO,                     KC_NO,   KC_NO,   KC_NO
```

**Comment by jackytrabbit** [comment]
Thanks guys, I appreciate!

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**